src/App.js

```diff
@@ -1,41 +1,78 @@
 import React from "react";
-import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";
+import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
 import Home from "./pages/Home/";
 import Favorites from "./pages/Favorites/";
 import Search from "./components/Search/";
 import SearchPage from "./pages/SearchPage/";
 import Book from "./pages/Book/";
 import "./App.css";
+import { StyledLink } from "./app.styles";
 
-export default function App() {
-  return (
-    <Router>
-      <div>
-        <nav>
-          <ul>
-            <li>
-              <Link to="/">Home</Link>
-            </li>
-            <li>
-              <Search />
-            </li>
-            <li>
-              <Link to="/favorites">★</Link>
-            </li>
-          </ul>
-        </nav>
+export default class App extends React.Component {
+  state = {
+    favorites: [],
+  };
 
-        <Switch>
-          <Route path="/favorites">
-            <Favorites />
-          </Route>
-          <Route exact path="/">
-            <Home />
-          </Route>
-          <Route component={Book} path="/book/:isbn"></Route>
-          <Route component={SearchPage} path="/search/:searchTerm"></Route>
-        </Switch>
-      </div>
-    </Router>
-  );
+  handleClick = (id) => {
+    const index = this.state.favorites.indexOf(id);
+    const { favorites } = this.state;
+    if (this.state.favorites.includes(id) && index !== -1) {
+      this.state.favorites.splice(index, 1);
+      localStorage.setItem("favorites", JSON.stringify(favorites));
+    } else {
+      this.state.favorites.push(id);
+      localStorage.setItem("favorites", JSON.stringify(favorites));
+    }
+  };
+
+  componentDidMount() {
+    this.setState({
+      favorites: JSON.parse(localStorage.getItem("favorites")) || [],
+    });
+  }
+
+  render() {
+    return (
+      <Router>
+        <div>
+          <nav>
+            <ul>
+              <li>
+                <StyledLink to="/">Home</StyledLink>
+              </li>
+              <li>
+                <Search />
+              </li>
+              <li>
+                <StyledLink to="/favorites" style={{fontSize:"25px"}}>★</StyledLink>
```

Comment:
no inline style

Reply:
you can use react-icons npm, it has all icons in the world, you can use a start outline and then a starfill if the user adds that book to favorites

Reply:
Thanks for the tip Vlad. I like those icons!